src/lava/proc/dense/models.py

```diff
@@ -103,8 +109,12 @@ def run_spk(self):
         # accumulation at timestep t-1. This prevents deadlocking in
         # networks with recurrent connectivity structures.
         self.a_out.send(self.a_buff)
-        s_in = self.s_in.recv()
-        a_accum = self.weights[:, s_in].sum(axis=1)
+        if self.use_graded_spike.item():
+            s_in = self.s_in.recv()
+            a_accum = self.weights.dot(s_in)
+        else:
+            s_in = self.s_in.recv().astype(bool)
+            a_accum = self.weights[:, s_in].sum(axis=1)
```

Comment:
See previous comment. Would prefer self.weights.dot(s_in) for all.

Reply:
ibid.